README.md

```diff
@@ -27,11 +33,83 @@ The GraphQL examples are:
 * **createDumpSanitisedDB_noArgs.gql**: Variables must be set on the environment; the person running the task has no control
 * **createDumpSanitisedDB_setDBVariables.gql**: Allows the person running the task to additionally choose the database to which they connect
 
-## Process
+Most of the variables are explained in the example GraphQL files, but one in 
+particular requires a better writeup.
+
+### BUILDER_BACKUP_IMAGE_NAME
+
+This is the name of the resulting image to build, without the tag (eg, for 
+dockerhub myproject/image, or for custom registry quay.io/myproject/image, can 
+also be a lpattern like 
+`${registry}/${organization}/${project}/${environment}/${service}-data`)",
+
+The default value is `${registry}/${organization}/${project}/${environment}/${service}-data`
+
+The variables available are as follows:
+* `${registry}`: `BUILDER_REGISTRY_HOST` (variable defined in Lagoon organisation/project/environment)
+* `${organization}`: `BUILDER_REGISTRY_ORGANIZATION` (variable defined in Lagoon organisation/project/environment)
+* `${project}`: The name of the Lagoon project
+* `${environment}`: The name of the Lagoon environment
+* `${service}`: The name of the Lagoon service in the docker-compose file
+* `${database}`: The name of the database.  `${database}` is munged so that:
+    1. Any special character not allowed in DockerHub repo names is removed (replaced with nothing), and
+    2. If there are two special characters in a row, the first is retained, and later ones are removed (also as per DockerHub repo name requirements)
+
+## The Images
+
+There are functionally three images we have to worry about:
+1. The Sanitiser Image: This is the image that dumps the sanitised database and builds the other images
+2. The Sanitised Builder Image: This builds the sanitised image
+3. The Sanitised Clean Image: This is the clean MariaDB image that the sanitised database is put in
+
+## Building the Sanitiser Image
+
+The image is built using the `Dockerfile`.  Some parts are written in Go.  
+
+### The Go Parts
+
+This uses the `Makefile` to build the parts of the image that are written in Go.  
+This is the variable calculation part of the process, and was rewritten in Go 
+so that tests could be applied.  Other files associated with this part of the 
+process are:
+* `cmd/main.go`
+* `go.mod`
+* `go.sum`
+* `main.go`
+* `internal/builder/builder.go`
+* `internal/builder/builder_test.go`: Tests for `internal/builder/builder.go`
+* `internal/builder/variables.go`
+* `internal/builder/variables_test.go`: Tests for `internal/builder/variables.go`
```

Comment:
This isn't really necessary to add to the readme. I'll leave it for now though.

Reply:
Not for you maybe, but for all those who aren't fluent in Go yet, I'd like to see them actually documented better.  I may get to it at some point.  Thanks for leaving it.